libs/SalesforceSwiftSDK/SalesforceSwiftSDK/Extensions/SalesforceSDKManagerExtensions.swift

```diff
@@ -24,23 +24,43 @@
  */
 import Foundation
 import SalesforceSDKCore
+import SmartSync
 import PromiseKit
 
-extension SalesforceSDKManager {
-    
-    static var Builder = SalesforceSDKManagerBuilder.self
+class SalesforceSwiftSDKManager : SmartSyncSDKManager {
+   
+    override init() {
+        super.init()
+    }
     
-    class func configure(config : @escaping (SFSDKAppConfig) -> Void ) -> SalesforceSDKManager.Type  {
-        config(SalesforceSDKManager.shared().appConfig!)
-        return SalesforceSDKManager.self
+    override var appType: SFAppType {
+        get {
+            return SFAppType.native
+        }
     }
+ 
+}
 
-    class SalesforceSDKManagerBuilder {
+extension SalesforceSwiftSDKManager {
```

Comment:
Why even have an extension? Why not put the new methods in the class definition itself?

Reply:
@wmathurin its quite common to group distinct functionality through extensions. Easier to maintain in the long run as we add more extensions to the same class.

Reply:
It's one thing to extend an existing class - here the class is defined right there and has pretty much nothing.